packages/api-headless-cms/src/crud/contentEntry.crud.ts

```diff
@@ -1089,16 +1096,25 @@ export const createContentEntryCrud = (params: CreateContentEntryCrudParams): Cm
             });
         }
     };
-    const deleteEntry: CmsEntryContext["deleteEntry"] = async (model, entryId) => {
+    const deleteEntry: CmsEntryContext["deleteEntry"] = async (model, id, force) => {
         const permission = await checkEntryPermissions({ rwd: "d" });
         await checkModelAccess(context, model);
 
-        const storageEntry = await storageOperations.entries.getLatestRevisionByEntryId(model, {
-            id: entryId
-        });
+        const storageEntry = (await storageOperations.entries.getLatestRevisionByEntryId(model, {
+            id
+        })) as CmsEntry;
 
-        if (!storageEntry) {
-            throw new NotFoundError(`Entry "${entryId}" was not found!`);
+        if (!storageEntry && !force) {
+            throw new NotFoundError(`Entry "${id}" was not found!`);
+        } else if (force) {
```

Comment:
#### 1. Can we add comments maybe here? For example, I'm not clear what's L1111 doing / what's the intention here.

#### 2. Also, if I understood correctly, here we do:

Step1: If there's no `storageEntry` and `force` is false, just throw.
Step2: If `force` is true, then we do additional steps with `await deleteEntryHelper ...`

What's weird to me here this `else if (force)`. It's only checking the value of `force`. Shouldn't the complete condition here be `!storageEntry && force` ? If so, maybe a small refactor could be done here as well.

Reply:
Changed according to the comment. I also added comments above this part of the code.